src/collator/collator.py

```diff
@@ -145,14 +146,18 @@ def _sample_batch(self, batch_size):
             s.append(
                 self._normalise_states(
                     self._pad(
```

Comment:
I can't see the implementation of `self._pad` but in general I would recommend using `pad_sequence` from Pytorch which is highly efficient because written in C++: https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pad_sequence.html

Reply:
@maxtaylordavies